### PR TITLE
Added support for ODT export.

### PR DESCRIPTION
**Pull request by LarsGit223**
I added support for ODT export, so the variables will also be seen if you export to ODT/OpenOffice Writer.

### CODE REVIEW

File: syntax.php [hunk 1]
```diff
@@ -98,6 +98,12 @@ function render($mode, &$renderer, $data) {
         } elseif ($mode == 'metadata') {
             if ($renderer->capture) $renderer->doc .= $meta;
             return true;
+        } elseif ($mode == 'odt') {
+            // prevent caching to ensure the included pages are always fresh
+            if ($nocache) $renderer->info['cache'] = false;
+
+            $renderer->doc .= $renderer->cdata ($xhtml);
```
Comment (michitux) [inline syntax.php]: Doesn't `$renderer->cdata()` already update `$renderer->doc`? At least in the xhtml renderer, `$renderer->cdata()` has no return value.

**Comment by michitux** [comment]
I suggest the following fix instead:

```diff
diff --git a/syntax.php b/syntax.php
index e7c3fea..d935768 100644
--- a/syntax.php
+++ b/syntax.php
@@ -86,20 +86,17 @@ class syntax_plugin_var extends DokuWiki_Syntax_Plugin {
                 break;
         }
 
-        // for XHTML output
-        if ($mode == 'xhtml') {
-            // prevent caching to ensure the included pages are always fresh
-            if ($nocache) $renderer->info['cache'] = false;
+        if ($mode == 'metadata') {
+            $renderer->cdata($meta);
+        } else {
+            $renderer->cdata($xhtml);
 
-            $renderer->doc .= hsc($xhtml);
-            return true;
-
-        // for metadata renderer
-        } elseif ($mode == 'metadata') {
-            if ($renderer->capture) $renderer->doc .= $meta;
-            return true;
+            if ($nocache) {
+                $renderer->nocache();
+            }
         }
-        return false;
+
+        return true;
     }
 }
 // vim:ts=4:sw=4:et:enc=utf-8:

```

Can you think of any problems that could arise from rendering variables in arbitrary renderers? I could think of problems with the [searchtext](https://www.dokuwiki.org/plugin:searchtext) plugin, but then again this plugin has problems with any kind of dynamic output.

**Comment by LarsGit223** [comment]
Please excuse the late reponse.
I tested the code you suggested. It works and is nicer than my code.

Export works with ODT, PDF and pure text right now.
CSV export gives just an empty file, just like other plugin not supporting it, looks alright.

Currently, I do not see any problems allowing rendering in general. Anyway, if problems are reported in the future, the code can be surrounded by a check for xhtml, odt, pdf and text easily.

So for my opinion you can go ahead and use that code you suggested.

Thanks and regards,
Lars

**Comment by LarsGit223** [comment]
There are some conflicts which are probably easier to resolve by creating a new pull-request. So I close this one.

**Comment by LarsGit223** [comment]
Forget it, plugin var already has got ODT support.
This just seems to be an old leftover so there is no requirement for a new pull-request.

**Pull request closed without merging**